fairscale/nn/pipe/pipe.py

```diff
@@ -123,18 +201,11 @@ def split_module(
     """
     balance = list(balance)
 
-    if len(module) != sum(balance):
-        raise BalanceError(
-            "module and sum of balance have different length "
-            f"(module: {len(module)}, sum of balance: {sum(balance)})"
-        )
-
-    if any(x <= 0 for x in balance):
-        raise BalanceError(f"all balance numbers must be positive integer (balance: {balance})")
+    check_balance(module, balance)
 
-    if len(balance) > len(devices):
+    if devices and len(balance) > len(devices):
```

Comment:
minor nit: we could pass `style` as an argument to this function and avoid using `devices` to compare? 

Same comment for `verify_splitting` method.

Reply:
this is hard because these functions aren't methods of Pipeline, I think in this case it's ok because it's literally just checking the value of devices, not branching on some more complex behaviour.

Reply:
Sounds good, thanks.